src/Microsoft.Health.Fhir.Api/Features/Audit/AuditHelper.cs

```diff
@@ -68,6 +77,13 @@ private void Log(AuditAction auditAction, HttpStatusCode? statusCode, HttpContex
             IFhirRequestContext fhirRequestContext = _fhirRequestContextAccessor.RequestContext;
 
             string auditEventType = fhirRequestContext.AuditEventType;
+            var customHeaders = _auditHeaderReader.Read(httpContext) as Dictionary<string, string>;
+
+            // check for custom audit headers in context-response
+            if (auditAction == AuditAction.Executed)
+            {
+                CheckForCustomAuditHeadersInResponse(httpContext, customHeaders);
+            }
```

Comment:
Put it under if statement below.
No need to do computation if they not gonna be used.